test/e2e/software_upgrade_e2e_signet_launch_test.go

```diff
@@ -23,17 +23,22 @@ func (s *SoftwareUpgradeSignetLaunchTestSuite) SetupSuite() {
 
 	btcHeaderGenesis, err := app.SignetBtcHeaderGenesis(app.NewTmpBabylonApp().AppCodec())
 	s.NoError(err)
-	s.configurer, err = configurer.NewSoftwareUpgradeConfigurer(s.T(), false, config.UpgradeSignetLaunchFilePath, []*btclighttypes.BTCHeaderInfo{btcHeaderGenesis})
+
+	cfg, err := configurer.NewSoftwareUpgradeConfigurer(s.T(), true, config.UpgradeSignetLaunchFilePath, []*btclighttypes.BTCHeaderInfo{btcHeaderGenesis})
```

Comment:
Is it a good idea to let `NewSoftwareUpgradeConfigurer` return the concrete `UpgradeConfigurer` rather than the interface?

Reply:
https://github.com/babylonlabs-io/babylon/pull/25/commits/746b5839f2edd6f95d13267a39089f07e0c9b57d